.github/workflows/sanity_checks.yml

```diff
@@ -21,6 +21,33 @@ jobs:
         run: |
           ./tools/sanity_checks.py
 
+  MSYS2:
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    strategy:
+      matrix:
+        sys: [ MINGW64, MINGW32, UCRT64, CLANG32, CLANG64 ]
```

Comment:
MSYS is missing here. MSYS lacks many packages. Upstream does not want MSYS to be some alternative to cygwin. Maybe better to keep it out. Not sure.